src/controllers/auth.js

```diff
@@ -0,0 +1,83 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const ErrorResponse = require('../utils/errorResponse');
+const User = require('../models/user');
+
+exports.signup = async (req, res, next) => {
+  const { name, surname, emailAddress, schoolName, password, confirmPassword } =
+    req.body;
+
+  try {
+    // Check if email is unique
+    let user = await User.findOne({ email: emailAddress });
+    if (user) {
+      return next(new ErrorResponse('Email already exists!', 400));
+    }
+
+    // Check password equality
+    if (password !== confirmPassword) {
+      return next(new ErrorResponse('Passwords do not match!', 400));
+    }
+
+    // no such user exists, create new one
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    user = await User.create({
+      firstName: name,
+      lastName: surname,
+      email: emailAddress,
+      schoolName,
+      password: hashedPassword,
+    });
+
+    // delete user's password before returning
+    user = user.toObject();
+    delete user.password;
+
+    return res.status(201).json({ success: true, data: user });
+  } catch (error) {
+    return res.status(500).json(error);
+  }
+};
+
+exports.signin = async (req, res, next) => {
+  const { email, password } = req.body;
+  try {
```

Comment:
same as previous note could u validate if req.body properties is exist or not here

Reply:
Validation added for `/signin`